docs/docs/spec.md

```diff
@@ -321,15 +321,15 @@ We provide two parameterless declaration annotations: `@NullMarked` and
 Our declaration annotations are specified to be *recognized* when applied to the
 locations listed below:
 
--   A *named* class.
+-   A *named* type.
 -   A package.
 -   A module (for `@NullMarked` only, not `@NullUnmarked`).
 -   A method or constructor.
 
-> *Not* a field or a record component.
+All locations that are not explicitly listed as recognized are unrecognized.
 
-If our declaration annotations appear in any other location, they have no
-meaning.
+> That is, they are *not* recognized on a field, a parameter, a local variable,
+> an annotation interface, a type parameter, or a record component declaration.
```

Comment:
```suggestion
> a type parameter, or a record component declaration.
```

I think we should leave out "an annotation interface," since our annotations are still recognized there by virtue of being recognized on _any_ class/type.